h2o-core/src/main/java/water/Scope.java

```diff
@@ -33,17 +34,12 @@
    *  enter call except for the listed Keys.
    *  @return Returns the list of kept keys. */
   static public Key[] exit(Key... keep) {
-    List<Key> keylist = new ArrayList<>();
-    if( keep != null )
-      for( Key k : keep ) if (k != null) keylist.add(k);
-    Object[] arrkeep = keylist.toArray();
-    Arrays.sort(arrkeep);
+    Set<Key> mustKeep = Sets.newHashSet(keep);
```

Comment:
I agree with Michal here: `new HashSet<>(keep)` is clear and understandable. At the same time for `Sets.newHashSet(keep)` I'd need to learn what that library is and what it does. It's a bigger cognitive load without any benefit that I can see...